File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/07 - Nested Conditional Statements Exercise/09. On Time for the Exam .py
exam_hour = int(input())
exam_minute = int(input())
actual_hour = int(input())
actual_minute = int(input())

exam_time = (exam_hour * 60) + exam_minute

actual_time = (actual_hour * 60) + actual_minute

difference_time = (abs(actual_time - exam_time))
difference_hour = difference_time // 60
difference_minute = difference_time % 60

difference_early = exam_time - 30

if actual_time == exam_time:
    print("On time")
elif actual_time >= difference_early and actual_time < exam_time:
    print("On time")
    print(f"{difference_minute} minutes before the start")
elif actual_time < difference_early:
    if difference_time < 60:
        print("Early")
        print(f"{difference_minute} minutes before the start")
    else:
        if difference_minute < 10:
            print("Early")
            print(f"{difference_hour}:0{difference_minute} hours before the start")
        else:
            print("Early")
            print(f"{difference_hour}:{difference_minute} hours before the start")
elif actual_time > exam_time:
    if difference_time < 60:
        print("Late")
        print(f"{difference_minute} minutes after the start")
    else:
        if difference_minute < 10:
            print("Late")
            print(f"{difference_hour}:0{difference_minute} hours after the start")
        else:
            print("Late")
            print(f"{difference_hour}:{difference_minute} hours after the start")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/10. УПРАЖНЕНИЕ ПО-СЛОЖНИ ПРОВЕРКИ/09. On Time for the Exam.js
// This is probably, most likly, for sure the ugliest TASK SoftUni has...!

function onTimeForTheExam(input) {
    let hourExam = Number(input.shift());
    let minuteExam = Number(input.shift());
    let hourArrival = Number(input.shift());
    let minuteArrival = Number(input.shift());

    let differenceMinutes = ((hourExam * 60) + minuteExam) - ((hourArrival * 60) + minuteArrival);

    let hh = Math.abs(~~(differenceMinutes / 60));
    let mm = Math.abs(differenceMinutes % 60);
    if (mm == 0) {mm = `00`}
    let time = 0;
    if (hh == 0) {
        time = `${mm} minutes`;
    } else {
        if (Number(mm) <= 9 && Number(mm) != 0) {mm = `0${mm}`}
        time = `${hh}:${mm} hours`;
    }

    let beforeOrAfter = '';

    if (differenceMinutes >= 0 && differenceMinutes <= 30) {
        console.log('On time');
        beforeOrAfter = 'before';
    } else if (differenceMinutes > 30) {
        console.log('Early');
        beforeOrAfter = 'before';
    } else if (differenceMinutes < 0) {
        console.log('Late');
        beforeOrAfter = 'after';
    }

    if (Math.abs(differenceMinutes) > 0) {
        // Only if difference is bigger then one minute it will print more:
        console.log(`${time} ${beforeOrAfter} the start`);
    }
}

onTimeForTheExam(['9', '30', '9', '50']);  //Expected Output:
// Late
// 20 minutes after the start

onTimeForTheExam(['9', '00', '10', '30']);  //Expected Output:
// Late
// 1:30 hours after the start

onTimeForTheExam(['10', '00', '10', '00']);  //Expected Output:
// On time

onTimeForTheExam(['9', '00', '8', '30']);  //Expected Output:
// On time
// 30 minutes before the start

onTimeForTheExam(['14', '00', '13', '55']);  //Expected Output:
// On time
// 5 minutes before the start

onTimeForTheExam(['11', '30', '10', '55']);  //Expected Output:
// Early
// 35 minutes before the start

onTimeForTheExam(['16', '00', '15', '00']);  //Expected Output:
// Early
// 1:00 hours before the start

onTimeForTheExam(['11', '30', '8', '12']);  //Expected Output:
// Early
// 3:18 hours before the start

onTimeForTheExam(['11', '30', '12', '29']);  //Expected Output:
// Late
// 59 minutes after the start
